### ISSUE TITLE
Darkflow support for the VOC standard

### ISSUE THREAD

**Issue by 4F2E4A2E** [thtrieu/darkflow#406]
Hello there @thtrieu!

Thank you very much for this amazing repo. It has been a lot of fun working with darknet and darkflow and you made it all happen! 👍 

I am sure that, only in repo and issues management a lot of hours have flown, so thank you sir in many ways, i will try to keep it short:

What i want to learn and would love to undestand is how **VOC** is _really_ being integrated or regarded into [darknet](https://github.com/pjreddie/darknet) &/ **darkflow**.

**[VOC-Standard](http://www.frontiersincomputervision.com/slides/FCV_Dataset_Zisserman_2.pdf) / VOC-Dataset folder structure:**
```
-Annotations
-ImageSets
--Action
--Layout
--Main
--Segmentation
-JPEGImages
-SegmentationClass
-SegmentationObject
```

To my best knowledge: The current source code [1] in darkflow &/ [darknet](https://github.com/pjreddie/darknet) does not contain any references or is in any way processing the VOC-Standard in something else other than
* -Annotations (class / label annotation) and
* -JPEGImages (images)

If this is correct, than a enormous poetential in the training &/ recognition is ... lost? Not being used? Processed? Not to mention the annotation files itself, which contains flags like:
* [truncated](https://github.com/thtrieu/darkflow/search?utf8=%E2%9C%93&q=truncated&type=)
* occluded
* difficult, etc

Are they those information being processed? If yes how!? There seams to be nothing on this in the source code.

I am hoping that at _least_ **SegmentationClasses** and **SegmentationObjects** are being somehow involved and being processed by the model alias yolo-voc.weight files. 

_I kindly ask you to totally dismantle this guess or to enlighten us. That would be really great and helpfull!_ 

### This way:
* darkflow would still rock
* contributors could extend the functionality in order to fully support the VOC-Standard 
* knowing the limitations, datasets would be prepared much faster and slicker regarding the VOC-features supported by darkflow only

1:
* https://github.com/pjreddie/darknet/search?utf8=%E2%9C%93&q=truncated&type=
* https://github.com/thtrieu/darkflow/blob/master/darkflow/defaults.py
* https://github.com/thtrieu/darkflow/search?utf8=%E2%9C%93&q=truncated&type=

Thank you in advance!

**Comment by 4F2E4A2E** [comment]
Any hint?

**Comment by rlan** [comment]
TLDR I think darkflow only supports object detection eg no segmentation
AFAIK.

Here is how I understood the history of darkflow. It is a port of darknet
which is the code base for the paper by Redmon et al, YOLO. The paper topic
is on object detection.

**Comment by 4F2E4A2E** [comment]
It seams that [darknet](https://github.com/pjreddie/darknet/search?utf8=%E2%9C%93&q=segmentation&type=) _does_ make use of the segmentated images containing the ground truth. 
Maybe this is only not yet implemented &/ configured on darkflow?
